src/config.coffee

```diff
@@ -665,12 +665,23 @@ class Config
   #
   # Returns an {Object} eg. `{type: 'integer', default: 23, minimum: 1}`.
   # Returns `null` when the keyPath has no schema specified.
+  # Returns `false` when the key-path is not accessible from the root schema.
   getSchema: (keyPath) ->
     keys = splitKeyPath(keyPath)
     schema = @schema
     for key in keys
-      break unless schema?
-      schema = schema.properties?[key]
+      if schema.type is 'object'
+        childSchema = schema.properties?[key]
+        unless childSchema?
+          if isPlainObject(schema.additionalProperties)
+            childSchema = schema.additionalProperties
+          else if schema.additionalProperties is false
+            return false
+          else
+            return null
```

Comment:
I agree that the `null/false` thing is weird. But I feel like there's something intuitive about  `::getSchema` returning the same thing that was passed to `::setSchema`, as opposed to an object with additional structure. Maybe we could think of another way to encode the difference between "not accessible" and "whatever value you want" though.

How about we change `null` to represent "not accessible", and in the case where any value is allowed (e.g. `atom.config.getSchema("editor.unknownKey")`), we return a new schema like `{type: "any"}` that corresponds to no validation whatsoever.